### PR TITLE
Feature: quarto directives

### PR DESCRIPTION
**Pull request by cscheid**
Currently supported directives:

- `<include/>`

As it stands, this generates identical output on `quarto-web` aside from hashes and modified-by fields.

TODO:
- [x] remove `keep-if` xml directive
- [x] revert break-quarto-md changes that involve xml syntax that's not self-closing

post-merge?
- [x] add `pagebreak` directive
- [x] turn `keep-if` to `.hidden when-format` ... in lua filters

### CODE REVIEW

File: src/core/handlers/base.ts [hunk 1]
```diff
@@ -234,7 +341,14 @@ export const baseHandler: LanguageHandler = {
     cells: QuartoMdCell[],
   ): MappedString[] {
     this.documentStart(handlerContext);
-    const result = cells.map((cell) => this.cell(handlerContext, cell));
+    const result = cells.map((cell) => {
+      return this.cell(
+        handlerContext,
+        cell,
+        // FIXME this should get the project+document options as well.
+        mergeConfigs(this.defaultOptions ?? {}, cell.options ?? {}),
```
Comment (jjallaire) [inline src/core/handlers/base.ts:349]: Do you want to fix this before we merge?

File: src/core/handlers/keep-if.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  languageName: "keep-if",
+
+  defaultOptions: {},
+
+  cell(
+    handlerContext: LanguageCellHandlerContext,
+    cell: QuartoMdCell,
+  ) {
+    // FIXME there's probably a more elegant way to do this...
```
Comment (jjallaire) [inline src/core/handlers/keep-if.ts:34]: Let's discuss approach to this. We definitely need to aggregate as is being done here and we _may_ want to support some different macros as we do for. yaml validation.

File: src/core/lib/parse-directive-tag.ts [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  "tt",
+  "u",
+  "ul",
+  "var",
+  "video",
+  "wbr",
+  "xmp",
+]);
+
+export function isDirectiveTag(str: string) {
```
Comment (jjallaire) [inline src/core/lib/parse-directive-tag.ts:167]: If we don't match a directive, do we just assume it's an html custom tag (i.e. component)?  (seems like we should as if we don't that would preclude use of these tags)

File: src/core/handlers/base.ts (hunk 1 above)
Comment (cscheid) [reply src/core/handlers/base.ts:349]: Yes, sounds good. (In full disclosure, the same problem currently exists for cell handlers on `main`)

File: src/core/lib/parse-directive-tag.ts (hunk 3 above)
Comment (cscheid) [reply src/core/lib/parse-directive-tag.ts:167]: Yes, if it doesn't match a directive tag, then `breakQuartoMd` just adds the line to the markdown cell. And if the xml tag is well formed, then the only way `isDirectiveTag` returns true is if the tag name is an HTML element.

**Pull request merged**